contrib/long_distance_matching/main-ldm.c

```diff
@@ -0,0 +1,269 @@
+// TODO: file size must fit into a U32
```

Comment:
You only need one copy of main, since the versions expose the same functions.

Another option, which might help testing, is to make a struct like

```c
typedef struct {
    size_t (*compress)(void const* src, size_t srcSize, void* dst, size_t maxDstSize);
    size_t (*decompress)(void const* src, size_t srcSize, void* dst, size_t maxDstSize);
    char const* name;
} compressor;
```

Then each version implements a `static` compression function, and uses a shared decompression function, and at the bottom of the source file they write

```c
compressor const ldmv1_compressor = {
    .compress = LDMv1_compress,
    .decompress = LDM_decompress,
    .name = "LDMv1",
};
```

Then in a common header, have an array of compressors

```c
extern const compressor ldmv1_compressor;
extern const compressor ldmv2_compressor;
const compressor compressors[] = {
  ldmv1_compressor,
  ldmv2_compressor,
};
```

Then you can accept a "name" argument on the command line, and loop through the compressors until you find one that matches the name, and use that one. You could also benchmark them all in one command.